File: frontend/services/ai/adapters/GeminiAdapter.ts
/**
 * Gemini适配器实现
 * 支持文本生成、图片生成、音频处理、视频生成
 */

import { BaseAdapter } from '../base/BaseAdapter';
import {
  AIProvider,
  TextGenerationRequest,
  TextGenerationResponse,
  TextGenerationChunk,
  ImageGenerationRequest,
  ImageGenerationResponse,
  TextToSpeechRequest,
  SpeechToTextRequest,
  AudioResponse,
  VideoGenerationRequest,
  VideoGenerationResponse,
  TokenUsage,
} from '../types';
import { AIConfigManager } from '../config';

/**
 * Gemini适配器
 */
export class GeminiAdapter extends BaseAdapter {
  private readonly defaultBaseUrl = 'https://generativelanguage.googleapis.com/v1beta';

  constructor(apiKey?: string) {
    super(apiKey);
    this.baseUrl = this.defaultBaseUrl;
  }

  getProviderType(): AIProvider {
    return 'gemini';
  }

  supportsTextGeneration(): boolean {
    return true;
  }

  supportsImageGeneration(): boolean {
    return true;
  }

  supportsTextToSpeech(): boolean {
    return true;
  }

  supportsSpeechToText(): boolean {
    return true;
  }

  supportsVideoGeneration(): boolean {
    return true;
  }

  getSupportedModels(capability: 'text' | 'image' | 'audio' | 'video'): string[] {
    const models: Record<string, string[]> = {
      text: [
        'gemini-2.0-flash-exp',
        'gemini-1.5-pro',
        'gemini-1.5-flash',
        'gemini-pro',
      ],
      image: [
        'imagen-3.0-generate-001',
        'imagen-2',
      ],
      audio: [
        'gemini-2.0-flash-exp',
      ],
      video: [
        'veo-2',
      ],
    };
    return models[capability] || [];
  }

  async generateText(request: TextGenerationRequest): Promise<TextGenerationResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('gemini', 'text');
    const url = `${this.baseUrl}/models/${model}:generateContent?key=${this.apiKey}`;

    // 构建请求体
    const requestBody: any = {
      contents: [],
    };

    // 添加系统指令
    if (request.systemInstruction) {
      requestBody.systemInstruction = {
        parts: [{ text: request.systemInstruction }],
      };
    }

    // 添加对话历史或当前提示
    if (request.messages && request.messages.length > 0) {
      requestBody.contents = request.messages.map(msg => ({
        role: msg.role === 'assistant' ? 'model' : 'user',
        parts: [{ text: msg.content }],
      }));
    } else {
      requestBody.contents = [{
        role: 'user',
        parts: [{ text: request.prompt }],
      }];
    }

    // 添加生成配置
    requestBody.generationConfig = {
      temperature: request.temperature ?? 0.7,
      maxOutputTokens: request.maxTokens ?? 2048,
    };

    try {
      const response = await this.makeRequest<any>(url, {
        method: 'POST',
        body: JSON.stringify(requestBody),
      });

      // 解析响应
      const content = response.candidates?.[0]?.content?.parts?.[0]?.text || '';
      const usage = response.usageMetadata ? {
        inputTokens: response.usageMetadata.promptTokenCount || 0,
        outputTokens: response.usageMetadata.candidatesTokenCount || 0,
        totalTokens: response.usageMetadata.totalTokenCount || 0,
      } : undefined;

      return {
        content,
        provider: 'gemini',
        model,
        usage,
        finishReason: response.candidates?.[0]?.finishReason,
      };
    } catch (error) {
      throw new Error(`Gemini text generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async generateTextStream(
    request: TextGenerationRequest,
    onChunk: (chunk: TextGenerationChunk) => void
  ): Promise<void> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('gemini', 'text');
    const url = `${this.baseUrl}/models/${model}:streamGenerateContent?key=${this.apiKey}`;

    // 构建请求体（与generateText相同）
    const requestBody: any = {
      contents: [],
    };

    if (request.systemInstruction) {
      requestBody.systemInstruction = {
        parts: [{ text: request.systemInstruction }],
      };
    }

    if (request.messages && request.messages.length > 0) {
      requestBody.contents = request.messages.map(msg => ({
        role: msg.role === 'assistant' ? 'model' : 'user',
        parts: [{ text: msg.content }],
      }));
    } else {
      requestBody.contents = [{
        role: 'user',
        parts: [{ text: request.prompt }],
      }];
    }

    requestBody.generationConfig = {
      temperature: request.temperature ?? 0.7,
      maxOutputTokens: request.maxTokens ?? 2048,
    };

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(requestBody),
      });

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${await response.text()}`);
      }

      const reader = response.body?.getReader();
      const decoder = new TextDecoder();

      if (!reader) {
        throw new Error('Failed to get response reader');
      }

      let buffer = '';
      let fullContent = '';

      while (true) {
        const { done, value } = await reader.read();
        if (done) break;

        buffer += decoder.decode(value, { stream: true });
        const lines = buffer.split('\n');
        buffer = lines.pop() || '';

        for (const line of lines) {
          if (line.trim() === '') continue;
          
          try {
            const data = JSON.parse(line);
            const chunkText = data.candidates?.[0]?.content?.parts?.[0]?.text || '';
            
            if (chunkText) {
              fullContent += chunkText;
              onChunk({
                content: chunkText,
                done: false,
              });
            }

            // 检查是否完成
            if (data.candidates?.[0]?.finishReason) {
              const usage = data.usageMetadata ? {
                inputTokens: data.usageMetadata.promptTokenCount || 0,
                outputTokens: data.usageMetadata.candidatesTokenCount || 0,
                totalTokens: data.usageMetadata.totalTokenCount || 0,
              } : undefined;

              onChunk({
                content: '',
                done: true,
                usage,
              });
            }
          } catch (e) {
            // 忽略解析错误，继续处理下一行
          }
        }
      }
    } catch (error) {
      throw new Error(`Gemini stream generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async generateImage(request: ImageGenerationRequest): Promise<ImageGenerationResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('gemini', 'image');
    
    // Gemini 使用 generateContent API 生成图片
    const url = `${this.baseUrl}/models/${model}:generateContent?key=${this.apiKey}`;

    // 计算宽高比
    const width = request.width || 1024;
    const height = request.height || 1024;
    let aspectRatio = '1:1';
    if (width > height) {
      aspectRatio = `${Math.round(width / height)}:1`;
    } else if (height > width) {
      aspectRatio = `1:${Math.round(height / width)}`;
    }

    const requestBody = {
      contents: [{
        parts: [{
          text: request.prompt,
        }],
      }],
      generationConfig: {
        imageConfig: {
          aspectRatio: aspectRatio,
        },
      },
    };

    try {
      const response = await this.makeRequest<any>(url, {
        method: 'POST',
        body: JSON.stringify(requestBody),
      });

      const images: Array<{ url?: string; base64?: string }> = [];
      
      // 解析响应中的图片数据
      const candidates = response.candidates || [];
      for (const candidate of candidates) {
        const parts = candidate.content?.parts || [];
        for (const part of parts) {
          if (part.inlineData) {
            // Gemini 返回 base64 数据
            const base64 = part.inlineData.data;
            const mimeType = part.inlineData.mimeType || 'image/png';
            const dataUrl = `data:${mimeType};base64,${base64}`;
            
            images.push({
              base64: dataUrl,
            });
          }
        }
      }

      // 如果请求多张图片，需要多次调用（Gemini API 一次只能生成一张）
      const numberOfImages = request.numberOfImages || 1;
      if (numberOfImages > 1 && images.length < numberOfImages) {
        // 生成剩余的图片
        for (let i = images.length; i < numberOfImages; i++) {
          try {
            const additionalResponse = await this.makeRequest<any>(url, {
              method: 'POST',
              body: JSON.stringify(requestBody),
            });
            
            const additionalCandidates = additionalResponse.candidates || [];
            for (const candidate of additionalCandidates) {
              const parts = candidate.content?.parts || [];
              for (const part of parts) {
                if (part.inlineData) {
                  const base64 = part.inlineData.data;
                  const mimeType = part.inlineData.mimeType || 'image/png';
                  const dataUrl = `data:${mimeType};base64,${base64}`;
                  images.push({ base64: dataUrl });
                  break;
                }
              }
            }
          } catch (error) {
            console.warn(`[GeminiAdapter] Failed to generate additional image ${i + 1}:`, error);
          }
        }
      }

      return {
        images: images.slice(0, numberOfImages),
        provider: 'gemini',
        model,
        usage: {
          imagesGenerated: images.length,
        },
      };
    } catch (error) {
      throw new Error(`Gemini image generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async textToSpeech(request: TextToSpeechRequest): Promise<AudioResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['text']);

    // TODO: 实现Gemini TTS
    throw new Error('Gemini TTS not implemented yet');
  }

  async speechToText(request: SpeechToTextRequest): Promise<AudioResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['audioFile']);

    // TODO: 实现Gemini STT
    throw new Error('Gemini STT not implemented yet');
  }

  async generateVideo(request: VideoGenerationRequest): Promise<VideoGenerationResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    // TODO: 实现Gemini视频生成
    throw new Error('Gemini video generation not implemented yet');
  }
}

File: backend/src/main/java/com/heartsphere/aiagent/adapter/GeminiAdapter.java
package com.heartsphere.aiagent.adapter;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.heartsphere.aiagent.dto.request.*;
import com.heartsphere.aiagent.dto.response.*;
import com.heartsphere.aiagent.exception.AIServiceException;
import com.heartsphere.aiagent.util.StreamResponseHandler;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.*;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.*;

/**
 * Google Gemini 适配器实现
 * 通过 HTTP 调用 Google Gemini API
 * 
 * @author HeartSphere
 * @version 1.0
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class GeminiAdapter implements ModelAdapter {
    
    @Value("${spring.ai.gemini.api-key:}")
    private String defaultApiKey;
    
    @Value("${spring.ai.gemini.base-url:https://generativelanguage.googleapis.com/v1beta}")
    private String baseUrl;
    
    private final RestTemplate restTemplate;
    private final ObjectMapper objectMapper;
    
    // 支持的文本模型
    private static final List<String> TEXT_MODELS = Arrays.asList(
        "gemini-2.0-flash-exp", "gemini-1.5-pro", "gemini-1.5-flash", "gemini-pro"
    );
    
    // 支持的图片模型
    private static final List<String> IMAGE_MODELS = Arrays.asList(
        "imagen-3.0-generate-001"
    );
    
    @Override
    public String getProviderType() {
        return "gemini";
    }
    
    @Override
    public boolean supportsTextGeneration() {
        return true;
    }
    
    @Override
    public boolean supportsImageGeneration() {
        return true;
    }
    
    @Override
    public boolean supportsTextToSpeech() {
        return false; // Gemini 暂不支持 TTS
    }
    
    @Override
    public boolean supportsSpeechToText() {
        return false; // Gemini 暂不支持 STT
    }
    
    @Override
    public boolean supportsVideoGeneration() {
        return true; // 支持 Veo 模型
    }
    
    @Override
    public TextGenerationResponse generateText(TextGenerationRequest request) {
        try {
            log.debug("Gemini文本生成请求: model={}, prompt={}", request.getModel(), request.getPrompt());
            
            // 获取 API key（优先从请求中获取，否则使用默认配置）
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("Gemini API key 未配置");
            }
            
            String model = request.getModel() != null ? request.getModel() : "gemini-2.0-flash-exp";
            String url = String.format("%s/models/%s:generateContent?key=%s", baseUrl, model, apiKey);
            
            // 构建请求体
            Map<String, Object> requestBody = buildGeminiRequest(request);
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("Gemini API 调用失败: " + response.getStatusCode());
            }
            
            // 解析响应
            return parseGeminiResponse(response.getBody(), request);
            
        } catch (Exception e) {
            log.error("Gemini文本生成失败", e);
            throw new AIServiceException("Gemini文本生成失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public void generateTextStream(TextGenerationRequest request, 
                                   StreamResponseHandler<TextGenerationResponse> handler) {
        try {
            log.debug("Gemini流式文本生成请求: model={}", request.getModel());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                handler.handle(null, true);
                throw new AIServiceException("Gemini API key 未配置");
            }
            
            String model = request.getModel() != null ? request.getModel() : "gemini-2.0-flash-exp";
            String url = String.format("%s/models/%s:streamGenerateContent?key=%s", baseUrl, model, apiKey);
            
            // 构建请求体
            Map<String, Object> requestBody = buildGeminiRequest(request);
            
            // 发送流式请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            // 注意：这里简化实现，实际应该使用 SSE 或 WebClient 处理流式响应
            // 暂时使用同步方式，后续可以优化
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                handler.handle(null, true);
                throw new AIServiceException("Gemini API 调用失败");
            }
            
            // 解析流式响应（简化实现）
            parseGeminiStreamResponse(response.getBody(), request, handler);
            
        } catch (Exception e) {
            log.error("Gemini流式文本生成失败", e);
            handler.handle(null, true);
        }
    }
    
    @Override
    public ImageGenerationResponse generateImage(ImageGenerationRequest request) {
        try {
            log.debug("Gemini图片生成请求: model={}, prompt={}", request.getModel(), request.getPrompt());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("Gemini API key 未配置");
            }
            
            String model = request.getModel() != null ? request.getModel() : "imagen-3.0-generate-001";
            String url = String.format("%s/models/%s:predict?key=%s", baseUrl, model, apiKey);
            
            // 构建请求体
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("prompt", request.getPrompt());
            if (request.getWidth() != null) requestBody.put("width", request.getWidth());
            if (request.getHeight() != null) requestBody.put("height", request.getHeight());
            // 默认生成1张图片
            requestBody.put("numImages", 1);
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("Gemini 图片生成 API 调用失败");
            }
            
            // 解析响应
            return parseImageResponse(response.getBody(), request);
            
        } catch (Exception e) {
            log.error("Gemini图片生成失败", e);
            throw new AIServiceException("Gemini图片生成失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public AudioResponse textToSpeech(AudioRequest request) {
        throw new AIServiceException("Gemini 暂不支持文本转语音");
    }
    
    @Override
    public AudioResponse speechToText(AudioRequest request) {
        throw new AIServiceException("Gemini 暂不支持语音转文本");
    }
    
    @Override
    public VideoGenerationResponse generateVideo(VideoGenerationRequest request) {
        try {
            log.debug("Gemini视频生成请求: prompt={}", request.getPrompt());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("Gemini API key 未配置");
            }
            
            String model = request.getModel() != null ? request.getModel() : "veo-2";
            String url = String.format("%s/models/%s:generateVideo?key=%s", baseUrl, model, apiKey);
            
            // 构建请求体
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("prompt", request.getPrompt());
            if (request.getDuration() != null) requestBody.put("duration", request.getDuration());
            if (request.getResolution() != null) requestBody.put("resolution", request.getResolution());
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("Gemini 视频生成 API 调用失败");
            }
            
            // 解析响应
            return parseVideoResponse(response.getBody(), request);
            
        } catch (Exception e) {
            log.error("Gemini视频生成失败", e);
            throw new AIServiceException("Gemini视频生成失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public List<String> getSupportedModels(String capability) {
        switch (capability.toLowerCase()) {
            case "text":
                return TEXT_MODELS;
            case "image":
                return IMAGE_MODELS;
            case "video":
                return Arrays.asList("veo-2");
            default:
                return new ArrayList<>();
        }
    }
    
    /**
     * 获取 API key（优先从请求中获取，否则使用默认配置）
     * 注意：这里需要从 TextGenerationRequest 中获取 API key
     * 可以通过扩展请求对象或使用 ThreadLocal 传递
     */
    private String getApiKey(TextGenerationRequest request) {
        // TODO: 从管理后台配置中获取 API key
        // 当前先使用默认配置
        return defaultApiKey;
    }
    
    private String getApiKey(ImageGenerationRequest request) {
        // TODO: 从管理后台配置中获取 API key
        return defaultApiKey;
    }
    
    private String getApiKey(VideoGenerationRequest request) {
        // TODO: 从管理后台配置中获取 API key
        return defaultApiKey;
    }
    
    /**
     * 构建 Gemini API 请求体
     */
    private Map<String, Object> buildGeminiRequest(TextGenerationRequest request) {
        Map<String, Object> requestBody = new HashMap<>();
        List<Map<String, Object>> contents = new ArrayList<>();
        
        // 添加系统指令（如果有）
        if (request.getSystemInstruction() != null && !request.getSystemInstruction().isEmpty()) {
            Map<String, Object> systemMessage = new HashMap<>();
            systemMessage.put("role", "user");
            Map<String, Object> systemPart = new HashMap<>();
            systemPart.put("text", request.getSystemInstruction());
            systemMessage.put("parts", Arrays.asList(systemPart));
            contents.add(systemMessage);
        }
        
        // 添加历史消息
        if (request.getMessages() != null && !request.getMessages().isEmpty()) {
            for (TextGenerationRequest.Message msg : request.getMessages()) {
                Map<String, Object> message = new HashMap<>();
                message.put("role", "user".equals(msg.getRole()) ? "user" : "model");
                Map<String, Object> part = new HashMap<>();
                part.put("text", msg.getContent());
                message.put("parts", Arrays.asList(part));
                contents.add(message);
            }
        }
        
        // 添加当前提示词
        if (request.getPrompt() != null && !request.getPrompt().isEmpty()) {
            Map<String, Object> userMessage = new HashMap<>();
            userMessage.put("role", "user");
            Map<String, Object> part = new HashMap<>();
            part.put("text", request.getPrompt());
            userMessage.put("parts", Arrays.asList(part));
            contents.add(userMessage);
        }
        
        requestBody.put("contents", contents);
        
        // 添加生成配置
        Map<String, Object> generationConfig = new HashMap<>();
        if (request.getTemperature() != null) {
            generationConfig.put("temperature", request.getTemperature());
        } else {
            generationConfig.put("temperature", 0.7);
        }
        if (request.getMaxTokens() != null) {
            generationConfig.put("maxOutputTokens", request.getMaxTokens());
        }
        requestBody.put("generationConfig", generationConfig);
        
        return requestBody;
    }
    
    /**
     * 解析 Gemini API 响应
     */
    private TextGenerationResponse parseGeminiResponse(JsonNode response, TextGenerationRequest request) {
        TextGenerationResponse result = new TextGenerationResponse();
        result.setProvider(getProviderType());
        result.setModel(request.getModel() != null ? request.getModel() : "gemini-2.0-flash-exp");
        
        try {
            // 提取内容
            JsonNode candidates = response.get("candidates");
            if (candidates != null && candidates.isArray() && candidates.size() > 0) {
                JsonNode candidate = candidates.get(0);
                JsonNode content = candidate.get("content");
                if (content != null) {
                    JsonNode parts = content.get("parts");
                    if (parts != null && parts.isArray() && parts.size() > 0) {
                        JsonNode part = parts.get(0);
                        String text = part.get("text").asText();
                        result.setContent(text);
                    }
                }
                
                // 提取完成原因
                if (candidate.has("finishReason")) {
                    result.setFinishReason(candidate.get("finishReason").asText());
                }
            }
            
            // 提取 Token 使用量
            JsonNode usageMetadata = response.get("usageMetadata");
            if (usageMetadata != null) {
                TextGenerationResponse.TokenUsage usage = new TextGenerationResponse.TokenUsage();
                if (usageMetadata.has("promptTokenCount")) {
                    usage.setInputTokens(usageMetadata.get("promptTokenCount").asInt());
                }
                if (usageMetadata.has("candidatesTokenCount")) {
                    usage.setOutputTokens(usageMetadata.get("candidatesTokenCount").asInt());
                }
                if (usageMetadata.has("totalTokenCount")) {
                    usage.setTotalTokens(usageMetadata.get("totalTokenCount").asInt());
                }
                result.setUsage(usage);
            }
            
        } catch (Exception e) {
            log.warn("解析 Gemini 响应失败", e);
            result.setContent("");
        }
        
        return result;
    }
    
    /**
     * 解析 Gemini 流式响应
     */
    private void parseGeminiStreamResponse(JsonNode response, TextGenerationRequest request,
                                          StreamResponseHandler<TextGenerationResponse> handler) {
        try {
            // 简化实现：解析流式响应
            // 实际应该处理多个 chunk
            JsonNode candidates = response.get("candidates");
            if (candidates != null && candidates.isArray()) {
                for (JsonNode candidate : candidates) {
                    JsonNode content = candidate.get("content");
                    if (content != null) {
                        JsonNode parts = content.get("parts");
                        if (parts != null && parts.isArray()) {
                            for (JsonNode part : parts) {
                                if (part.has("text")) {
                                    String text = part.get("text").asText();
                                    TextGenerationResponse chunk = new TextGenerationResponse();
                                    chunk.setProvider(getProviderType());
                                    chunk.setModel(request.getModel() != null ? request.getModel() : "gemini-2.0-flash-exp");
                                    chunk.setContent(text);
                                    handler.handle(chunk, false);
                                }
                            }
                        }
                    }
                }
            }
            
            // 完成
            TextGenerationResponse finalResponse = new TextGenerationResponse();
            finalResponse.setProvider(getProviderType());
            finalResponse.setModel(request.getModel() != null ? request.getModel() : "gemini-2.0-flash-exp");
            handler.handle(finalResponse, true);
            
        } catch (Exception e) {
            log.error("解析 Gemini 流式响应失败", e);
            handler.handle(null, true);
        }
    }
    
    /**
     * 解析图片生成响应
     */
    private ImageGenerationResponse parseImageResponse(JsonNode response, ImageGenerationRequest request) {
        ImageGenerationResponse result = new ImageGenerationResponse();
        result.setProvider(getProviderType());
        result.setModel(request.getModel() != null ? request.getModel() : "imagen-3.0-generate-001");
        
        try {
            List<ImageGenerationResponse.Image> images = new ArrayList<>();
            JsonNode generatedImages = response.get("generatedImages");
            if (generatedImages != null && generatedImages.isArray()) {
                for (JsonNode image : generatedImages) {
                    ImageGenerationResponse.Image img = new ImageGenerationResponse.Image();
                    if (image.has("url")) {
                        img.setUrl(image.get("url").asText());
                    }
                    if (image.has("base64")) {
                        img.setBase64(image.get("base64").asText());
                    }
                    images.add(img);
                }
            }
            result.setImages(images);
            
            ImageGenerationResponse.Usage usage = new ImageGenerationResponse.Usage();
            usage.setImagesGenerated(images.size());
            result.setUsage(usage);
            
        } catch (Exception e) {
            log.warn("解析 Gemini 图片响应失败", e);
            result.setImages(new ArrayList<>());
        }
        
        return result;
    }
    
    /**
     * 解析视频生成响应
     */
    private VideoGenerationResponse parseVideoResponse(JsonNode response, VideoGenerationRequest request) {
        VideoGenerationResponse result = new VideoGenerationResponse();
        result.setProvider(getProviderType());
        result.setModel(request.getModel() != null ? request.getModel() : "veo-2");
        
        try {
            if (response.has("videoUrl")) {
                result.setVideoUrl(response.get("videoUrl").asText());
                result.setStatus("completed");
            } else if (response.has("taskId")) {
                result.setVideoId(response.get("taskId").asText());
                result.setStatus("processing");
            }
        } catch (Exception e) {
            log.warn("解析 Gemini 视频响应失败", e);
            result.setStatus("failed");
        }
        
        return result;
    }
}

